src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/UpgradeToRegexGeneratorAnalyzerTests.cs

```diff
@@ -69,13 +69,19 @@ public async Task NoDiagnosticForConstructorWithTimeout(string test)
         }
 
         [Fact]
-        public async Task NoDiagnosticForTopLevelStatements()
+        public async Task TopLevelStatements()
         {
             string test = @"using System.Text.RegularExpressions;
+Regex r = [|new Regex("""")|];";
+            string fixedCode = @"using System.Text.RegularExpressions;
+Regex r = MyRegex();
```

Comment:
Can we use `ConstructRegexInvocation` and turn this into a theory to make sure we have coverage for both Regex constructor and the static methods?